library/src/blas2/trmv_template.hpp

```diff
@@ -0,0 +1,123 @@
+/* ************************************************************************
```

Comment:
Hm, why are we splitting this into another file? Previously we've had rocblas_func.hpp define the rocblas_func_template() function which had the generic strided_batched/batched/non-batched implementation.

Reply:
That's what I have done, rocblas_trmv.hpp rocblas_trmv_stride_batched.hpp etc.
trmv_template is only a intermediate routine.

Reply:
I see. I'm just not sure where we ended up with the conversation that was had a month or two ago about the "C++ interface", since most other routines don't have a C++ rocblas_batched_template/rocblas_strided_batched_template function I believe.
Not a big deal, just think consistency between routines is important. I know gemv is an outlier right now for having a gemv_device.hpp file. Iirc we started moving away from this, although I know there was some contention.

Reply:
Postponed discussion since it is a valid point.